[BUG] Error after upgrading to hypothesis 6.49

**Checklist**

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)

**Describe the bug**
An error occurs when running a simple test with library `hypothesis` version 6.49.1. The test is the basic:
```
@schema.parametrize()
def test_api(case):
    response = case.call_asgi()
    case.validate_response(response)
```
No change in the API specs.

### Successful run
```
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: ..., configfile: pytest.ini
plugins: anyio-3.5.0, schemathesis-3.15.6, ***hypothesis-6.48.1***, subtests-0.5.0, nbmake-1.3.0
collected 20 items                                                                                                                                                                                

tests/ ....................    
```
### After upgrading
 After upgrading `hypothesis` to the latest minor:
```
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: ..., configfile: pytest.ini
plugins: anyio-3.5.0, schemathesis-3.15.6, ***hypothesis-6.49.1***, subtests-0.5.0, nbmake-1.3.0
collected 0 items / 1 error                                                                                                                                                                       

============================================================================================= ERRORS ==============================================================================================
_________________________________________________________________________ ERROR collecting tests/client/test_by_schema.py _________________________________________________________________________
../.venv/lib/python3.8/site-packages/pluggy/_hooks.py:265: in __call__
    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
../.venv/lib/python3.8/site-packages/pluggy/_manager.py:80: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
../.venv/lib/python3.8/site-packages/schemathesis/extra/pytest_plugin.py:211: in pytest_pycollect_makeitem
    outcome.force_result(create(SchemathesisCase, parent=collector, test_function=obj, name=name))
../.venv/lib/python3.8/site-packages/schemathesis/extra/pytest_plugin.py:38: in create
    return cls.from_parent(*args, **kwargs)  # type: ignore
../.venv/lib/python3.8/site-packages/_pytest/nodes.py:264: in from_parent
    return cls._create(parent=parent, **kw)
../.venv/lib/python3.8/site-packages/_pytest/nodes.py:140: in _create
    return super().__call__(*k, **kw)
../.venv/lib/python3.8/site-packages/schemathesis/extra/pytest_plugin.py:78: in __init__
    failing_test = validate_given_args(test_function, given_args, given_kwargs)
../.venv/lib/python3.8/site-packages/schemathesis/utils.py:446: in validate_given_args
    return is_invalid_test(func, argspec, args, kwargs)  # type: ignore
../.venv/lib/python3.8/site-packages/hypothesis/core.py:278: in is_invalid_test
    params = list(original_sig.parameters.values())
E   AttributeError: 'FullArgSpec' object has no attribute 'parameters'
```

**Expected behavior**
Test should work with the latest hypothesis version

see at https://github.com/HypothesisWorks/hypothesis/issues/3397

I found everything to work ok if pinned to `hypothesis == 6.48.3` or lower.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -174,3 +174,35 @@ jobs:
       uses: codecov/codecov-action@v3.0.0
       with:
         file: ./coverage.xml
+
+  hypothesis-6-48:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04]
+        tox_job:
+          - py3-hypothesis-6-48
+
+    name: ${{ matrix.os }}/tests_${{ matrix.tox_job }}
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3.0.0
+      with:
+        fetch-depth: 1
+
+    - uses: actions/setup-python@v3.1.0
+      with:
+        python-version: 3.8
+
+    - run: pip install tox coverage poetry
+
+    - name: Run ${{ matrix.tox_job }} tox job
+      run: tox -e ${{ matrix.tox_job }}
+
+    - run: coverage combine
+    - run: coverage report
+    - run: coverage xml -i
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v3.0.0
+      with:
+        file: ./coverage.xml

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -11,6 +11,7 @@ Changelog
 **Fixed**
 
 - Compatibility with ``hypothesis>=6.49``. `#1538`_
+- Handling of ``unittest.case.SkipTest`` emitted by newer Hypothesis versions.
 
 **Removed**
 

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -8,6 +8,10 @@ Changelog
 
 - Report uploading to Schemathesis.io via the ``--report`` CLI option.
 
+**Fixed**
+
+- Compatibility with ``hypothesis>=6.49``. `#1538`_
+
 **Removed**
 
 - Previously, data was uploaded to Schemathesis.io when the proper credentials were specified. This release removes this behavior.
@@ -2955,6 +2959,7 @@ Deprecated
 .. _0.3.0: https://github.com/schemathesis/schemathesis/compare/v0.2.0...v0.3.0
 .. _0.2.0: https://github.com/schemathesis/schemathesis/compare/v0.1.0...v0.2.0
 
+.. _#1538: https://github.com/schemathesis/schemathesis/issues/1538
 .. _#1526: https://github.com/schemathesis/schemathesis/issues/1526
 .. _#1518: https://github.com/schemathesis/schemathesis/issues/1518
 .. _#1514: https://github.com/schemathesis/schemathesis/issues/1514

--- a/src/schemathesis/_compat.py
+++ b/src/schemathesis/_compat.py
@@ -1,4 +1,5 @@
 # pylint: disable=ungrouped-imports
+import hypothesis
 import hypothesis_jsonschema._from_schema
 import jsonschema
 import werkzeug
@@ -25,6 +26,11 @@ try:
     from hypothesis.utils.conventions import InferType
 except ImportError:
     InferType = type(...)
+
+if version.parse(hypothesis.__version__) >= version.parse("6.49"):
+    from hypothesis.internal.reflection import get_signature
+else:
+    from inspect import getfullargspec as get_signature
 
 
 def _get_format_filter(

--- a/src/schemathesis/runner/impl/core.py
+++ b/src/schemathesis/runner/impl/core.py
@@ -2,6 +2,7 @@
 import logging
 import threading
 import time
+import unittest
 import uuid
 from contextlib import contextmanager
 from types import TracebackType
@@ -298,6 +299,10 @@ def run_test(  # pylint: disable=too-many-locals
             result.mark_skipped()
         else:
             status = Status.success
+    except unittest.case.SkipTest:
+        # Newer Hypothesis versions raise this exception if no tests were executed
+        status = Status.skip
+        result.mark_skipped()
     except CheckFailed:
         status = Status.failure
     except NonCheckError:

--- a/src/schemathesis/utils.py
+++ b/src/schemathesis/utils.py
@@ -38,7 +38,7 @@ from requests.exceptions import InvalidHeader  # type: ignore
 from requests.utils import check_header_validity  # type: ignore
 from werkzeug.wrappers import Response as BaseResponse
 
-from ._compat import InferType, JSONMixin
+from ._compat import InferType, JSONMixin, get_signature
 from .constants import USER_AGENT, DataGenerationMethod
 from .exceptions import UsageError
 from .types import DataGenerationMethodInput, Filter, GenericTest, NotSet, RawAuth
@@ -442,8 +442,8 @@ def merge_given_args(func: GenericTest, args: Tuple, kwargs: Dict[str, Any]) -> 
 
 
 def validate_given_args(func: GenericTest, args: Tuple, kwargs: Dict[str, Any]) -> Optional[Callable]:
-    argspec = getfullargspec(func)
-    return is_invalid_test(func, argspec, args, kwargs)  # type: ignore
+    signature = get_signature(func)
+    return is_invalid_test(func, signature, args, kwargs)  # type: ignore
 
 
 def compose(*functions: Callable) -> Callable:

--- a/test/specs/openapi/test_serializing.py
+++ b/test/specs/openapi/test_serializing.py
@@ -128,6 +128,7 @@ def assert_generates(testdir, raw_schema, expected, parameter):
 
     testdir.make_test(
         f"""
+import json
 from urllib.parse import quote, unquote
 
 def chunks(items, n):
@@ -151,6 +152,11 @@ class Prefixed:
         return self.prepare(instance) == self.prepare(other)
 
 
+class JSONString(Prefixed):
+    def prepare(self, value):
+        return json.loads(unquote(value))
+
+
 class CommaDelimitedObject(Prefixed):
     def prepare(self, value):
         if not value:
@@ -169,7 +175,6 @@ class DelimitedObject(Prefixed):
         return dict(item.split("=") for item in items)
 
 @schema.parametrize()
-@settings(phases=[Phase.explicit])
 def test_(request, case):
     request.config.HYPOTHESIS_CASES += 1
     assert case.{attribute} in {repr(expected)}

--- a/test/test_lazy.py
+++ b/test/test_lazy.py
@@ -34,7 +34,7 @@ def test_(request, case):
     )
     result = testdir.runpytest("-v", "-s")
     # Then settings should be applied to the test
-    result.assert_outcomes(passed=1)
+    result.assert_outcomes(passed=1, skipped=1)
     result.stdout.re_match_lines([r"test_with_settings.py::test_ PASSED", r".*1 passed"])
     result.stdout.re_match_lines([r"Hypothesis calls: 0$"])
 

--- a/tox.ini
+++ b/tox.ini
@@ -30,7 +30,7 @@ commands =
 [testenv:py3-hypothesis-6-48]
 commands =
   poetry install -v
-  poetry run pip install "hypothesis<6.48"
+  poetry run pip install "hypothesis<6.49"
   coverage run --source=schemathesis -m pytest {posargs:} test
 
 [testenv:coverage-report]

--- a/tox.ini
+++ b/tox.ini
@@ -4,6 +4,7 @@ envlist =
   py{36,37,38,39,310},
   py3-pytest53,
   py3-pytest6,
+  py3-hypothesis-6-48,
   coverage-report
 
 [testenv]
@@ -24,6 +25,12 @@ commands =
 commands =
   poetry install -v
   poetry run pip install "pytest<7" "pytest-subtests<0.7.0"
+  coverage run --source=schemathesis -m pytest {posargs:} test
+
+[testenv:py3-hypothesis-6-48]
+commands =
+  poetry install -v
+  poetry run pip install "hypothesis<6.48"
   coverage run --source=schemathesis -m pytest {posargs:} test
 
 [testenv:coverage-report]

